chain/chain/src/resharding/manager.rs

```diff
@@ -169,6 +170,23 @@ impl ReshardingManager {
         store_update.commit()
     }
 
+    /// TODO(resharding): Remove once proper solution for negative refcounts is implemented.
+    fn duplicate_nodes_at_split_boundary<'a>(
```

Comment:
Why do we need this change?
It won't preserve all intermediary nodes , only a subset of them, right?

Reply:
Yes, only a subset. It includes intermediary nodes for singletons, like `PromiseYieldIndices`. Also, it is possible that we should duplicate all these nodes at split boundary anyway.

Reply:
Oh I see, this makes for an easy way to access some intermediary nodes. 

I can't figure out with certainty if this touches the entire trie path for all singletons, it looks like yes to me, but @shreyan-gupta what's your take ?

Reply:
Yeah, this should be able to touch all singleton nodes.